src/pages/index.js

```diff
@@ -100,36 +146,95 @@ function fillProfileForm() {
     descInput.value = info.description;
 }
 
-function updateInfo(event, data) {
-    userInfo.setUserInfo({
+async function updateInfo(event, data) {
+    const newInfo = {
         name: data["name-input"],
         description: data["description-input"],
+    };
+    userInfo.setUserInfo(newInfo);
+    api.editProfile({
+        name: newInfo.name,
+        about: newInfo.description
     });
```

Comment:
Every server request in `index.js` should look something like this:

```
api.someMethod()
.then(res => {
// here is the response that should be used to update the DOM.
// also in `then` you should close the popup
})
.catch(err => // here you catch possible errors)
.finally(() => // here you bring the default button text back )
``` 

Please, correct it in the whole code